pandas/core/frame.py

```diff
@@ -8201,7 +8201,7 @@ def stack(self, level: Level = -1, dropna: bool = True):
 
     def explode(
         self,
-        column: Scalar | tuple | list[Scalar | tuple],
+        column: IndexLabel | tuple,
```

Comment:
tuple is hashable so can remove that too.

Reply:
Hm forgot that.Thx

Reply:
@simonjayhawkins got a follow up question: IndexLabel does include Series, DataFrame and arrays? So I think I misunderstood you in the other pr? We should annotate this with ``Hashable| tuple | list[Hashable| tuple]`` not IndexLabel, because arrays and so on are not allowed here

Reply:
IndexLabel does not yet include Series, DataFrame, Index, EAs etc since they do not satisfy typing.Seqence

For public methods, the function should be as permissible as possible and consistent with other methods accepting similar parameters.

If `explode` is more restrictive than other methods that accept a `columns` parameter, we should probably fix. (and maybe should not use the IndexLabel alias yet)

Reply:
So we should use IndexLabel always when we have a single column or a list of columns if possible? And when we have only a single column Hashable would be appropriate?

Reply:
yes for consistency in public methods. but the types should also reflect the types accepted by the method so where we have discrepancies, mypy should hightlight this as we add more types. Typing of the public api is more difficult, especially since we don't have many of the lower level functions typed.

for internal methods we can be less permissive and say only pass a list around. (The issue then becomes that the list is mutable, so function code could unintentionally change the contents and mypy would catch that.)

Reply:
yes of course, if its more strict we should define that. Thanks for the explanation, this helps a lot.